tests/system/test_aggregation_query.py

```diff
@@ -530,3 +530,141 @@ def test_aggregation_query_with_nested_query_multiple_filters(
+    )
+    agg_query.count()
+    agg_query.sum("appearances")
+    agg_query.avg("appearances")
+    iterator = agg_query.fetch()
+    # should have plan_summary but no execution_stats
+    stats = iterator.explain_metrics
+    assert isinstance(stats, ExplainMetrics)
+    assert isinstance(stats.plan_summary, PlanSummary)
+    assert len(stats.plan_summary.indexes_used) > 0
+    # execution_stats should not be present
+    with pytest.raises(QueryExplainError) as excinfo:
+        stats.execution_stats
+    assert "execution_stats not available" in str(excinfo.value)
+    # should have no results
+    assert len(list(iterator)) == 0
+
+
+@pytest.mark.parametrize("database_id", [None, _helpers.TEST_DATABASE], indirect=True)
+def test_aggregation_query_explain_analyze(
+    aggregation_query_client, nested_query, database_id
+):
+    """
+    When explain_options(analyze=True) is set, iterator should contain explain_metrics field
+    with plan_summary and execution_stats
+
+    Should not be present until iterator is exhausted
+    """
+    from google.cloud.datastore.query_profile import QueryExplainError
+    from google.cloud.datastore.query_profile import ExplainOptions
+    from google.cloud.datastore.query_profile import ExplainMetrics
+    from google.cloud.datastore.query_profile import ExecutionStats
+    from google.cloud.datastore.query_profile import PlanSummary
+
+    expected_error = "explain_metrics not available until query is complete."
+    agg_query = aggregation_query_client.aggregation_query(
+        nested_query, explain_options=ExplainOptions(analyze=True)
+    )
+    agg_query.count()
+    agg_query.sum("appearances")
+    agg_query.avg("appearances")
+    iterator = agg_query.fetch()
+    # explain_metrics isn't present until iterator is exhausted
+    with pytest.raises(QueryExplainError) as excinfo:
+        iterator.explain_metrics
+    assert expected_error in str(excinfo.value)
+    # exhaust the iterator
+    results = list(iterator)
+    num_results = len(results)
+    assert num_results > 0
+    stats = iterator.explain_metrics
+    assert isinstance(stats, ExplainMetrics)
+    # verify plan_summary
+    assert isinstance(stats.plan_summary, PlanSummary)
+    assert len(stats.plan_summary.indexes_used) > 0
+    assert (
+        stats.plan_summary.indexes_used[0]["properties"]
+        == "(appearances ASC, __name__ ASC)"
+    )
+    assert stats.plan_summary.indexes_used[0]["query_scope"] == "Includes ancestors"
+    # verify execution_stats
+    assert isinstance(stats.execution_stats, ExecutionStats)
+    assert stats.execution_stats.results_returned == num_results
+    assert stats.execution_stats.read_operations == num_results
+    duration = stats.execution_stats.execution_duration.total_seconds()
+    assert duration > 0
+    assert duration < 1  # we expect a number closer to 0.05
+    assert isinstance(stats.execution_stats.debug_stats, dict)
```

Comment:
I would also assert on one or two of the debug_stats values

Reply:
Do we have any guarantees about any fields that will always be present?

Reply:
As we discussed, I just added checks for "billing_details", "documents_scanned", "index_entries_scanned", like [you have in java](https://github.com/googleapis/java-datastore/blame/main/google-cloud-datastore/src/test/java/com/google/cloud/datastore/it/ITDatastoreTest.java#L668)